### PR TITLE
Use ruff for linting and fix (almost) all lints.

### PR DESCRIPTION
**Pull request by ocefpaf**
Note that I explicitly disabled bugbear, that found some legit bugs, b/c I want to fix those in a separate PR. They will demand a more carefull review than the automated changes here. I also skipped  the 8 errors for the same reason.

```
compliance_checker/cf/cf_1_8.py:176:24: F821 Undefined name `error_msgs`
compliance_checker/cf/cf_1_8.py:658:9: F811 Redefinition of unused `check_geometry` from line 615
compliance_checker/cf/cf_1_8.py:790:7: F811 Redefinition of unused `PointGeometry` from line 509
compliance_checker/cf/cf_1_8.py:854:7: F811 Redefinition of unused `PolygonGeometry` from line 578
compliance_checker/cf/cf_base.py:188:19: UP030 Use implicit references for positional format fields
compliance_checker/cf/cf_base.py:687:17: UP030 Use implicit references for positional format fields
compliance_checker/cf/cf_base.py:1286:7: F811 Redefinition of unused `CFNCCheck` from line 1229
compliance_checker/tests/test_ioos_profile.py:947:9: F811 Redefinition of unused `test_check_contributor_role_and_vocabulary` from line 423
```

@benjwadams this PR is quite big b/c we have been ignoring linting for a while now. Hopefully, if this is accepted. We will be more careful as we move forward.

PS: merge #1009 first. I'll rebase this one and move to #1008 after.

### CODE REVIEW

File: compliance_checker/cf/cf.py [hunk 1]
```diff
@@ -1,40 +1,4 @@
...
-    grid_mapping_dict16,
-    grid_mapping_dict17,
-    horizontal_datum_names17,
-    prime_meridian_names17,
-)
 
 # Version specific checkers organized in other modules
-from compliance_checker.cf.cf_1_6 import CF1_6Check
-from compliance_checker.cf.cf_1_7 import CF1_7Check
-from compliance_checker.cf.cf_1_8 import CF1_8Check
```
Comment (ocefpaf) [inline compliance_checker/cf/cf.py:40]: This module is only imports. Do we need to keep this for compatibility? Should we rever this change?

Comment (ocefpaf) [reply compliance_checker/cf/cf.py]: I restored this one and added a skip for it.

**Comment by ocefpaf** [comment]
@benjwadams this one is ready for review. It is quite disruptive due to the number of files it touches but hopefully we can enforce the lints failures in the near-future.

**Comment by benjwadams** [comment]
Looks good in general, but just one question -- trailing commas on function parameters is the preferred style?

**Comment by ocefpaf** [comment]
> trailing commas on function parameters is the preferred style?

I don't like it... But yeah. Let me see if that if we can customize that. I need to rebase this anyway.

**Comment by ocefpaf** [comment]
@benjwadams to avoid conflicts headaches let's try to merge some low hanging fruit PRs first. How about #1016, #1014, #1013, and #1003?

**Comment by benjwadams** [comment]
Could you rebase and possibly rerun `ruff`?  Was trying to work through merge conflicts, but I think I'm going to make a mistake trying to work through these.

**Comment by ocefpaf** [comment]
> Could you rebase and possibly rerun `ruff`? Was trying to work through merge conflicts, but I think I'm going to make a mistake trying to work through these.

I'll probably issue a new one b/c it is easier. Let's sync up next week and do this.

**Comment by ocefpaf** [comment]
Closing this in favor of #1020

**Pull request closed without merging**